autoload/lsp/omni.vim

```diff
@@ -80,10 +77,11 @@ function! lsp#omni#complete(findstart, base) abort
 endfunction
 
 function! s:get_insertion_point(item, current_line, typed_pattern) abort
-    if !has_key(a:item, 'user_data')
-        let l:insert_start = -1
-    else
-        let l:insert_start = get(json_decode(a:item['user_data']), s:user_data_insert_start_key, -1)
+    let l:insert_start = -1
+
+    let l:user_data = lsp#omni#get_managed_user_data_from_completed_item(a:item)
+    if has_key(l:user_data, 'completion_item') && has_key(l:user_data['completion_item'], 'textEdit')
+        let l:insert_start = l:user_data['completion_item']['textEdit']['range']['start']['character']
```

Comment:
if you sync with latest master I add `lsp#utils#position#_lsp_to_vim(l:user_data['completion_item']['textEdit']['range']['start']`. This is less error prone and auto converts LSP chars to vim so you don't have to remember `-1` and `+1`.